### DISCUSSION TITLE
Release 2.2.0

### DISCUSSION THREAD

**Discussion by ChenglongMa** [ChenglongMa/zoplicate#18] [General]
[![zotero target version](https://img.shields.io/badge/For_Zotero-7-green?style=flat-square&logo=zotero&logoColor=CC2936)](https://www.zotero.org)
[![Version for Zotero 7](https://img.shields.io/github/package-json/v/ChenglongMa/zoplicate)](https://github.com/ChenglongMa/zoplicate/releases/tag/2.2.0)
[![Downloads for Zotero 7](https://img.shields.io/github/downloads/ChenglongMa/zoplicate/2.2.0/total)](https://github.com/ChenglongMa/zoplicate/releases/tag/2.2.0)

# Changelog

In this version, we have made the following changes:

1. ✨ **NEW!**: We have added **"Smart" Merge** functionality (Only available in Zotero 7 now).
   * When merging duplicate items, the function will fill in the missing information of the retained item with the duplicates.
   * Thanks [endfreude](https://github.com/endfreude)'s idea mentioned in [issue #15](https://github.com/ChenglongMa/zoplicate/issues/15).

<hr /><em>This discussion was created from the release <a href='https://github.com/ChenglongMa/zoplicate/releases/tag/2.2.0'>Release 2.2.0</a>.</em>

**Comment by Apollo7777777** [comment]
Hey, I found it can't work when these duplicate items are in different collection.

**Reply by ChenglongMa** [reply]
Hi @Apollo7777777,

Thanks for your feedback. I will check this later.

Could you please add more details like when this issue occurred? When you importing new items or running the Bulk merge function?

Thanks!

**Reply by Apollo7777777** [reply]
I have some files before installing the plugin, I import duplicate entries in a new category after installing the plugin, and then I find that it can't detect that there are duplicate entries before... But now when I import new literature that is not in the library to test, there is no problem

**Reply by Apollo7777777** [reply]
The previous issue is temporarily resolved as I am ready to re-establish the database. But I have a suggestion: it's important to be able to mark how many duplicates there are after detecting duplicates, like Endnote, because sometimes I need to give accurate numbers to others, when I have a lot of duplicates

**Reply by ChenglongMa** [reply]
Hi @Apollo7777777,

Thanks for your valuable feedback and suggestions! 

Could you find the duplicate entries in different collections in the "Duplicate Items" pane?

Since I'm using [Zotero's native detection method](https://www.zotero.org/support/duplicate_detection), if you can find them in the "Duplicate Items" pane and Zoplicate can't detect it, it's a bug in that plugin. I'll go fix it.

![Duplicate Items Window](https://www.zotero.org/support/_media/duplicate_detection.png?w=706&tok=0bbc43)

Then, I will add the function you suggested.

Thank you so much!

Best,
Chenglong

**Comment by Apollo7777777** [comment]
It would be great if I checked a few categories and then the plugin could help me find out how many duplicates there were. Further, is it possible to move duplicates from the original category to other categories after finding duplicates, as in the case of endnotes? looking forward to your news

**Reply by ChenglongMa** [reply]
Thanks for this suggestion.

I will check the functionalities in Endnotes and see if I can migrate them into Zotero.

Thanks!

**Reply by Apollo7777777** [reply]
Thank you very much for your great work, if you have a new progress, please tell me

**Reply by ChenglongMa** [reply]
Hi @Apollo7777777,

Hope you are doing well.

I have finished the function to display the duplicate count, as shown below:

![Snipaste_2024-03-17_15-22-33](https://github.com/ChenglongMa/zoplicate/assets/33501679/1b5c2d03-9f55-46f3-b8e2-0050d79c9421)

It will show the "**unique/total**" count and a **tooltip**.

However, about the other function:

> move duplicates from the original category to other categories after finding duplicates

I checked the duplicate functions in Endnotes and these documents: [this one](https://libguides.federation.edu.au/c.php?g=831674&p=5937461) and [this one](https://library-guides.ucl.ac.uk/endnote/duplicates).

I didn't find any similar functions.
Could you please add more details about this?

Thanks!

Best,
Chenglong

**Comment by Apollo7777777** [comment]
By the way, I found that the count of duplicate entries is calculated for the entire library, when I create a new group library (not a group), under the duplicate entries in the sublist, it can't only count the duplicate entries in the group library, I wonder if you have a way to allow users to count only the duplicate entries under the library in the group library? Zotero can only have one library, and it is not possible to create different libraries like Endnote, so it is necessary to calculate duplicate entries separately for different groups of libraries, which will reduce interference between different groups of libraries.

**Reply by ChenglongMa** [reply]
Hi @Apollo7777777,

I'm not sure if I understood your questions correctly.

I tried to reproduce your comments as follows:

![Snipaste_2024-03-14_15-19-43](https://github.com/ChenglongMa/zoplicate/assets/33501679/de4582d7-d16b-4e16-8169-9920b7a978d7)

As shown in the screenshot, there are 3 libraries (the default "My Library", and two Group Libraries "Zoplicate" and "Zoplicate2").

I did the following steps:

1. I imported two duplicates to "Collection 1" in "My Library". Let's call them "A1" and "A2".
2. Then, I imported two different duplicates to "Foo" in "Zoplicate". Let's call them "B1" and "B2".

I found the "Duplicate Items" pane can show them separately and correctly, i.e.,

* "My Library"'s "Duplicate Items" pane only shows "A1" and "A2";
* "Zoplicate"'s "Duplicate Items" pane only shows "B1" and "B2", as shown in the screenshot;

Could you kindly let me know if I did it correctly or how to reproduce the issue?

Thanks!

**Reply by Apollo7777777** [reply]
Thank you for the test, are your A1 and B1 the same entry?I just retested, in the case of A1 in my library, import A1 in the group library, no longer trigger the reminder of duplicate entries, thank you for your patience, it seems that my group library and my library are not interoperable, maybe I made a mistake before

**Reply by ChenglongMa** [reply]
Hi @Apollo7777777,

A1 and B1 are the same in my test. I removed the duplicates and retested again:

0. Remove all items in both "My Library" and "Zoplicate";
1. Imported A1 to "My Library";
2. Imported B1 (B1=A1) to "Zoplicate" => Didn't trigger the reminder;
3. Imported A2 (A2=A1) to "My Library" **again** => Triggered;
4. Merged the duplicates => Only one A1 remains in "My Library" and B1 is still in "Zoplicate";
5. Imported B2 (B2=A1) to "Zoplicate" **again** => Triggered the reminder;
6. Merged the duplicates => Only one A1 remains in "My Library" and only B1 remains in "Zoplicate";

Is this expected?

Thanks!

**Reply by Apollo7777777** [reply]
Yes, it's very nice. It seems that there was a problem with the data source and the entries I exported from Cochrane were duplicates themselves. OO

**Reply by Apollo7777777** [reply]
hhh, thank you very much. could I add your whatsapp account? it is not convenient to chat here

**Reply by ChenglongMa** [reply]
Oh, I didn't use Cochrane before. I guess the importing referred **the items in different group libraries** to **the same entry in the database**. Then, the operations on one are also synchronously mapped to the other.

I don't use WhatsApp too much. Let me set up one account :)

**Reply by ChenglongMa** [reply]
Hi @Apollo7777777,

I'm not sure if I am doing it correctly. 

Can you see the QR code?

Add me as a contact on WhatsApp. https://wa.me/qr/2DFRNWR6OKWHN1![shared_qr_code.png](https://github.com/ChenglongMa/zoplicate/assets/33501679/b81c4d76-5d80-4116-a570-186f88499ab7)

**Reply by ChenglongMa** [reply]
Seems not working...

Maybe we can chat in Discord. You can join the group via this link: https://discord.gg/e8jEBhut.

Thanks!
